### PR TITLE
gNOI-3.2: Skip test for no removable fabric card and line cards found

### PR DESCRIPTION
**Pull request by nicjiang2023**
Addressed comment: https://github.com/openconfig/featureprofiles/pull/2139#issuecomment-1750880594

### CODE REVIEW

File: feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go [hunk 1]
```diff
@@ -170,7 +169,7 @@ func TestLinecardReboot(t *testing.T) {
 		}
 	}
 	if removableLinecard == "" {
-		t.Fatalf("Component(lc).Removable().Get(t): got none, want non-empty")
+		t.Skipf("No removable line card found for the testing")
```
Comment (Ankur19) [inline feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go:172]: Having 0 removable linecards for a modular devce is an error.

Can you check if `arg.NumLinecards > 0` and then throw error if `removableLinecard == ""`

File: feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go [hunk 2]
```diff
@@ -241,7 +239,7 @@ func TestFabricReboot(t *testing.T) {
 		}
 	}
 	if removableFabric == "" {
-		t.Fatalf("Component(fabric).Removable().Get(t): got none, want non-empty")
+		t.Skipf("No fabric component found for the testing")
```
Comment (Ankur19) [inline feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go:242]: Same as above.

File: feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go (hunk 1 above)
Comment (nicjiang2023) [reply feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go]: Thanks for your comments. Updated script as below:
```go
		if *args.NumLinecards > 0 {
			t.Fatalf("No removable line card found for the testing on a modular device")
		} else {
			t.Skipf("No removable line card found for the testing")
		}
```

File: feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go (hunk 2 above)
Comment (nicjiang2023) [reply feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go]: Thanks for your comments. Updated script as below:
```go
		if *args.NumLinecards > 0 {
			t.Fatalf("No removable fabric component found for the testing on a modular device")
		} else {
			t.Skipf("No removable fabric component found for the testing")
		}
```

**Review by sachendras — APPROVED** [review]

**Comment by sachendras** [comment]
Approving this change. But an internal bug is required to cover for workflow changes to pass num_fabric arg.

**Review by sachendras — APPROVED** [review]

**Pull request merged**